internal/dag/builder.go

```diff
@@ -493,6 +484,37 @@ func (b *Builder) computeHTTPLoadBalancers() {
 	}
 }
 
+// mergeConditions merges any two conditions when they are delegated
+func mergeConditions(delegatedCondition, includeCondition *projcontour.Condition) *projcontour.Condition {
+	if delegatedCondition == nil {
+		return includeCondition
+	}
+
+	returnCondition := delegatedCondition.DeepCopy()
+	returnCondition.Prefix = delegatedCondition.Prefix + includeCondition.Prefix
+	return returnCondition
+}
+
+func (b *Builder) processIncludes(httplb *projcontour.HTTPLoadBalancer, host string, delegatedCondition *projcontour.Condition, enforceTLS bool) {
+	// Loop over and process all includes
+	for _, include := range httplb.Spec.Includes {
+		if delegatedHTTPLb, ok := b.Source.httploadbalancers[Meta{name: include.Name, namespace: include.Namespace}]; ok {
+			switch {
+			//	case ir.Spec.TCPProxy != nil && (passthrough || enforceTLS):
+			//		b.processTCPProxy(ir, nil, host)
+			case httplb.Spec.Routes != nil:
+				fmt.Println("processing an include: ", delegatedHTTPLb.Name)
+				b.processRoutes(delegatedHTTPLb, host, mergeConditions(delegatedCondition, &include.Condition), enforceTLS)
+			}
+
+			// Loop over any includes in the delegated httlb
+			if len(delegatedHTTPLb.Spec.Includes) > 0 {
```

Comment:
I think this needs some logic, similar to processIngressRoutes and processTCP that prevents it from including itself in a loop. We need to have this before this PR is landed because an infinite loop will stall the dag builder.

Reply:
thank you